docs/design/rescheduling.md

```diff
@@ -0,0 +1,104 @@
+* Queue Filter
+
+  `Queue filtering` will filter resources in specified queue. Then the rescheduling process only works on the result set.
+
+* Label Filter
+
+  `Label Filter` will filter pods with specified labels. Then the rescheduling process only works on the result set.
+
+### Rescheduling Strategy
+* OfflineOnly
+
+    `OfflineOnly`, abbreviated as `OLO`, will only pick out offline workloads, which are attached with annotation
+`preemptable: true`, and then reschedule them only.
+
+* LowPriorityFirst
+
+    `LowPriorityFirst`, abbreviated as `LPF`, will sort workloads by priority and then reschedule pods in ascending 
+order.
+
+* ShortLifeTimeFirst
+
+  `ShortLifeTimeFirst`, abbreviated as `SLTF`, will sort workloads by running time. Pods with the shortest lift time 
+will be rescheduled first. This strategy can make sure workloads with long task type goes healthily without interruptions.
+
+* BigObjectFirst
+
+    `BigObjectFirst`, abbreviated as `BOF`, will select workloads which request the most **dominate resource** and reschedule
+them first. It helps improve system throughout and avoid small workloads' starvation.
+
+* MoreReplicasFirst
+  
+    `MoreReplicasFirst`, abbreviated as `MRF`, will sort workloads by replicas number. Workloads with the most replicas
+will be rescheduled first. This strategy is friendly to `gang scheduling` for it will consider the `minAvailable` in 
+volcano jobs.
+
+### Metrics
+All the decisions made by rescheduling strategies will consider the metrics from `Prometheus`. Namely, Volcano will 
+list the real node resource utilization and pod distribution instead of request resource. Basically, usage of`CPU` and 
+`Memroy` will be collected. Other resource such as `GPU` can be extended.
+
+## Implementation(Beta)
+```yaml
+## Configuration Option 
+actions: "enqueue, allocate, backfill, shuffle"  ## add 'shuffle' at the end of the actions
+tiers:
+  - plugins:
+      - name: priority
+      - name: gang
+      - name: conformance
+      - name: rescheduling       ## rescheduling plugin
+        arguments:
+          interval: 5m
+          strategy:
+            - offlineOnly
+            - lowPriorityFirst
+          queueSelector:         ## optional, select workloads in specified queues as potential evictees. All queues by default.
+            - default
+            - test-queue
+          labelSelector:         ## optional, select workloads with specified labels as potential evictees. All labels by default.
+            business: offline
+            team: test
+  - plugins:
+      - name: overcommit
+      - name: drf
+      - name: predicates
+      - name: proportion
+      - name: nodeorder
+      - name: binpack
+```
+
+* Load and parse user configurations about rescheduling.
+* Update the cache by metrics collected by `Prometheus`.
+* Create a new action named "shuffle", implement the workflow above.
+* Create a new plugin named "rescheduling", implement the strategies above.
```

Comment:
What's your plan for dynamic scheduling and rescheduling. combine them in one plugin or two plugin?

Reply:
Will implement them as two plugins. Besides, the logic of part of functions for them is absolutely different.